pkg/helm/option.go

```diff
@@ -383,10 +389,94 @@ func WithMaxHistory(max int32) HistoryOption {
 	}
 }
 
-// NewContext creates a versioned context.
+// NewContext creates a versioned context with kubernetes client data.
 func NewContext() context.Context {
-	md := metadata.Pairs("x-helm-api-client", version.Version)
-	return metadata.NewContext(context.TODO(), md)
+	return metadata.NewContext(
+		context.TODO(),
+		metadata.Join(
+			metadata.New(
+				extractKubeConfig()),
+			metadata.New(map[string]string{
+				"x-helm-api-client": version.Version,
+			}),
+		),
+	)
+}
+
+func extractKubeConfig() map[string]string {
```

Comment:
pkg/helm should not parse kubeconfig, it should be passed into the context from cmd/helm